app/models/setting.rb

```diff
@@ -252,6 +256,16 @@ def self.column_check(opts)
 
   private
 
+  def validate_host_owner
+    owner_id = self.value
+    owner = User.find_by_id(owner_id.to_i) if owner_id =~ (/-Users\Z/)
```

Comment:
owner_id is a `type_and_id` value, therefore `to_i` casting takes only the integer part.